tsdb/meta_test.go

```diff
@@ -92,6 +93,38 @@ func Test_SeriesIDs_Reject(t *testing.T) {
 	}
 }
 
+func BenchmarkMeasurement_SeriesIDForExp(b *testing.B) {
+	m := tsdb.NewMeasurement("cpu")
+	for i := 0; i < 100000; i++ {
+		s := tsdb.NewSeries("cpu", models.Tags{models.Tag{
+			Key:   []byte("host"),
+			Value: []byte(fmt.Sprintf("host%d", i))}})
+		s.ID = uint64(i)
+		m.AddSeries(s)
+	}
+
+	if exp, got := 100000, len(m.SeriesKeys()); exp != got {
+		b.Fatalf("series count mismatch: exp %v got %v", exp, got)
+	}
+
+	stmt, err := influxql.NewParser(strings.NewReader(`SELECT * FROM cpu WHERE host =~ /host\d+/`)).ParseStatement()
```

Comment:
You should benchmark the other scenarios too. Your benchmark only tests one of the four scenarios. As mentioned in my previous comment, it's one of the two that would improve. I think if you tried the other two scenarios, you would have more memory allocations.